### PR TITLE
Add "automerge" option to MERGE_COMMIT_MESSAGE

### PR DESCRIPTION
**Pull request by leomp12**
### What

I'm providing a way to ignore PR title/description on auto merge commit message.

### Why

I was having troubles to [skip ci] on Netlify preview for Dependabot updates, but not [skip ci] on PR merged to master.
Commit messages on PR have `[skip ci]`, but merge commit shouldn't have.

#### Working example (Proposal)

https://github.com/ecomclub/storefront/commit/978e3b10a2738b98af3c59b4c2e805108cab4f63

#### Not working example (Current)

https://github.com/ecomclub/storefront/commit/de8ab04181654a78bba82c1bf86d56a1c77f78f8

> I've tried with _MERGE_METHOD_ squash and merge, and _MERGE_COMMIT_MESSAGE_ automatic, pull-request-title and pull-request-description, always ending with a [skip ci] somewhere on merge commit message :disappointed:

### CODE REVIEW

**Comment by pascalgn** [comment]
I just tried again with the latest version, and using `MERGE_COMMIT_MESSAGE: "pull-request-description"` works successfully for me, so only the PR body/description is used, not the PR title. Could you maybe check again?

**Comment by leomp12** [comment]
Hey @pascalgn,

You're right, but Dependabot ships the updated package changelog om description, and it may have a `[skip ci]` (in my case) somewhere.

So anyway imho would be good to have a new option with hardseted commit message.

**Comment by pascalgn** [comment]
Agree. I'm just against the `automerge` key, as it doesn't seem to be very useful for most other users.

How about instead we implement something where e.g. `MERGE_COMMIT_MESSAGE: "pull-request-description"` will use the PR body, as usual, but `MERGE_COMMIT_MESSAGE: "=Hello, world"` (prefix `=`) will use the literal text `Hello, world` as commit message.

WDYT?

**Comment by leomp12** [comment]
Yup. Sounds better, maybe we can also support `{number}` to replace by PR number on this option with `=` prefix, wdyt?

You want me to fix it on this PR?

**Comment by pascalgn** [comment]
Yes, good idea! But maybe let's use `{pullRequest.number}` as that matches the GH API more closely.

I thought a bit more about it and I think the `=` prefix isn't really necessary. So when the text is not `pull-request-title`,  `pull-request-description` etc., we can just treat it as the literal commit message (with placeholders).

Sure, if you have some time, feel free to give it a shot!

**Comment by leomp12** [comment]
Actually I thought there was a reason you were not accepting any arbitrary value in this field, as I did not get to read all the code...
Otherwise I would have suggested it this way before, certainly better than the mocked option I had added :grin:

File: lib/merge.js [hunk 1]
```diff
@@ -160,7 +160,10 @@ function getCommitMessage(mergeCommitMessage, pullRequest) {
   } else if (mergeCommitMessage === "pull-request-title-and-description") {
     return pullRequest.title + "\n\n" + pullRequest.body;
   } else {
-    throw new Error(`unknown commit message value: ${mergeCommitMessage}`);
+    ;['number', 'title', 'body'].forEach(prProp => {
```
Comment (pascalgn) [inline lib/merge.js]:
```suggestion
    ["number", "title", "body"].forEach(prProp => {
```

Possibly also change prProp to "property"

Comment (leomp12) [reply lib/merge.js]: To be honest, imo `property` is so much generic name, `prProp` sounds at least a bit more descriptive I think

File: README.md [hunk 2]
```diff
@@ -89,8 +89,10 @@ The following merge options are supported:
 - `MERGE_COMMIT_MESSAGE`: The commit message to use when merging the pull
   request into the base branch. Possible values are `automatic` (use GitHub's
   default message), `pull-request-title` (use the pull request's title),
-  `pull-request-description` (use the pull request's description), and
-  `pull-request-title-and-description`. The default value is `automatic`.
+  `pull-request-description` (use the pull request's description),
+  `pull-request-title-and-description` or a literal
+  value with optional placeholders (eg.: _Auto merge #{pullRequest.number}_).
```
[the remark this answers is not in the document]
Comment (leomp12) [reply README.md]: Right :+1:

File: lib/merge.js (hunk 1 above)
Comment (pascalgn) [reply lib/merge.js]: Fair enough!

File: lib/merge.js [hunk 3]
```diff
@@ -160,7 +160,10 @@ function getCommitMessage(mergeCommitMessage, pullRequest) {
   } else if (mergeCommitMessage === "pull-request-title-and-description") {
     return pullRequest.title + "\n\n" + pullRequest.body;
   } else {
-    throw new Error(`unknown commit message value: ${mergeCommitMessage}`);
+    ["number", "title", "body"].forEach(prProp => {
+      mergeCommitMessage.replace(new RegExp(`{pullRequest.${prProp}}`, 'g'), pullRequest[prProp]);
```
Comment (pascalgn) [inline lib/merge.js]: I just noticed this won't work, as `replace` doesn't change the string, the return value needs to be used!

Comment (leomp12) [reply lib/merge.js]: oops, you're right, my junior fault :grin:

Comment (leomp12) [reply lib/merge.js]: https://github.com/pascalgn/automerge-action/pull/45/commits/069a0ef207c6e3e387f2022d63fe6151765d54cd

**Comment by pascalgn** [comment]
@leomp12 Merged and released! Thanks for the PR! 👍

**Pull request merged**